File: src/main/java/tk/hiddenname/smarthome/service/hardware/impl/GPIOService.java
package tk.hiddenname.smarthome.service.hardware.impl;

import tk.hiddenname.smarthome.exception.GPIOBusyException;
import tk.hiddenname.smarthome.exception.PinSignalSupportException;

public interface GPIOService {

    void delete(Long id);

    void save(Long id, int gpioPin, boolean reverse) throws GPIOBusyException,
            PinSignalSupportException;

    void update(Long id, boolean reverse);
}

File: src/main/kotlin/tk/hiddenname/smarthome/service/hardware/impl/GPIOService.kt
package tk.hiddenname.smarthome.service.hardware.impl

import tk.hiddenname.smarthome.model.hardware.GPIO

interface GPIOService {

    fun delete(id: Long)

    fun save(id: Long, gpio: GPIO, reverse: Boolean)

    fun update(id: Long, reverse: Boolean)
}